src/main/java/com/app/folioman/config/db/FlexyPoolDataSourceConfig.java

```diff
@@ -0,0 +1,68 @@
+package com.app.folioman.config.db;
+
+import com.vladmihalcea.flexypool.FlexyPoolDataSource;
+import com.vladmihalcea.flexypool.adaptor.HikariCPPoolAdapter;
+import com.vladmihalcea.flexypool.config.FlexyPoolConfiguration;
+import com.vladmihalcea.flexypool.connection.ConnectionDecoratorFactoryResolver;
+import com.vladmihalcea.flexypool.metric.MetricsFactoryResolver;
+import com.vladmihalcea.flexypool.strategy.IncrementPoolOnTimeoutConnectionAcquisitionStrategy;
+import com.vladmihalcea.flexypool.strategy.RetryConnectionAcquisitionStrategy;
+import com.vladmihalcea.flexypool.strategy.UniqueNamingStrategy;
+import com.zaxxer.hikari.HikariDataSource;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import org.springframework.aop.scope.ScopedProxyUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.autoconfigure.orm.jpa.HibernateJpaAutoConfiguration;
+import org.springframework.context.annotation.Bean;
+
+@AutoConfiguration(
+        after = {DataSourceAutoConfiguration.class},
+        before = {HibernateJpaAutoConfiguration.class})
+@ConditionalOnClass({FlexyPoolDataSource.class, DataSource.class})
+@ConditionalOnBean(DataSource.class)
+public class FlexyPoolDataSourceConfig {
+
+    private static int maxOverflowPoolSize = 5;
+    private static int connectionAcquisitionThresholdMillis = 50;
+
+    @Bean
+    static FlexyPoolDataSourceBeanPostProcessor flexyPoolDataSourceBeanPostProcessor() {
+        return new FlexyPoolDataSourceBeanPostProcessor(maxOverflowPoolSize, connectionAcquisitionThresholdMillis);
+    }
+
+    private record FlexyPoolDataSourceBeanPostProcessor(
```

Comment:
@coderabbitai, how to handle this method effectively and trigger only when `FlexiPool` is in classpath

Reply:
@coderabbitai, any better alternative to define the code instead of current way?

Reply:
@coderabbitai, I am looking for a code like this only but when written code like this it is throwing error in `MFNavServiceImpl.java` that a bean with name `PlatformTransactionManager` should be created.

Reply:
@coderabbitai, can you please review both `LazyConnectionDataSourceProxyConfig` and `FlexyPoolDataSourceConfig` . Suggest best way to achieve the same

Reply:
@coderabbitai, the above code works but my intention is to let spring create datasource using autoconfiguration and then optionally wrap it to FlexydataSource if it is available in classpath . Then this datasource will be used for creating `ProxyDataSource` in `DataSourceObservationAutoConfiguration` . Finally use that `ProxyDataSource` to create `LazyConnectionDataSourceProxy`

Using above way I am leveraging spring boots auto configuration and injecting/modifying final datasource as per best practrices. Please suggest how to achive the same

Reply:
@coderabbitai, above code works with minor tweaks. Is it good practice for `LazyConnectionDataSourceProxyConfig` to implement `BeanFactoryPostProcessor` and writing `postProcessBeanFactory` as along with above suggested code, If there is better way, please suggest.

```java
 @Override
    public void postProcessBeanFactory(ConfigurableListableBeanFactory beanFactory) throws BeansException {
        // Retrieve the existing dataSource bean definition
        BeanDefinition originalBeanDefinition = beanFactory.getBeanDefinition("dataSource");

        // Check if the bean definition is an instance of AbstractBeanDefinition
        if (originalBeanDefinition instanceof AbstractBeanDefinition originalAbstractBeanDefinition) {
            // Clone the bean definition
            AbstractBeanDefinition clonedBeanDefinition = originalAbstractBeanDefinition.cloneBeanDefinition();
            // Modify the cloned bean definition
            clonedBeanDefinition.getPropertyValues().add("poolName", "jobrunr");
            // Register the cloned bean definition with a new name
            DefaultListableBeanFactory listableBeanFactory = (DefaultListableBeanFactory) beanFactory;
            listableBeanFactory.registerBeanDefinition("jobrunrDataSource", clonedBeanDefinition);
        } else {
            throw new IllegalStateException(
                    "Bean definition for 'dataSource' is not an instance of AbstractBeanDefinition");
        }
    }
```

Reply:
@coderabbitai, how to create a flexipooldatasource if it is created instead of normal datasource while creating 
`JobRunrDataSource`

Reply:
@coderabbitai, above code is not working with below error

Description:

Parameter 6 of constructor in com.app.folioman.mfschemes.service.MFNavServiceImpl required a bean of type 'org.springframework.transaction.PlatformTransactionManager' that could not be found.


Action:

Consider defining a bean of type 'org.springframework.transaction.PlatformTransactionManager' in your configuration.